plugins/Hosting/js/sagas/saga.js

```diff
@@ -204,6 +206,33 @@ function *requestDefaultSettingsSaga() {
 	}
 }
 
+// getHostStatus returns a promise that resolves with `true` if the provided
+// netaddress is connectable, or `false` if it is not connectable. The input
+// 'hostAddr' must be an address in the URL format, eg:
+// tcp://123.456.789.0:9982
+const getHostStatus = (hostAddr) => new Promise((resolve) => {
+	const hostUrl = url.parse(hostAddr)
+	const socket = net.connect(hostUrl.port, hostUrl.hostname)
+	socket.setTimeout(15e3) // 15 second host connectability timeout
```

Comment:
I'd aim for closer to 10 minutes. Host status isn't going to change faster than that, and it gives you more time to be certain that you are actually offline if nobody is interacting with you.